templates/CRM/common/importProgress.tpl

```diff
@@ -0,0 +1,81 @@
+ | Copyright CiviCRM LLC (c) 2004-2017                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+*}
+{literal}
+<script type="text/javascript">
+CRM.$(function($) {
+  $("form#Preview").on('submit', function (e) {
+    verify();
+  });
+  function setIntermediate() {
+    var dataUrl = "{/literal}{$statusUrl}{literal}";
+    $.getJSON(dataUrl, function(response) {
+      var dataStr = response.toString();
+      var result  = dataStr.split(",");
+      $("#intermediate").html(result[1]);
+      $("#importProgressBar .ui-progressbar-value").show();
+      if (result[0] < 100) {
+        $("#importProgressBar .ui-progressbar-value").animate({width: result[0] + "%"}, 500);
+        $("#status").text(result[0] + "% Completed");
+      }
+    });
+  }
+
+  function pollLoop() {
+    setIntermediate();
+    window.setTimeout(pollLoop, 10*1000); // 10 sec
+  }
+
+  function verify() {
+    if (!confirm('Backing up your database before importing is recommended, as there is no Undo for this. {/literal}{ts escape='js'}Are you sure you want to Import now{/ts}{literal}?')) {
+      return false;
+    }
+    $("#id-processing").show( ).dialog({
+      modal         : true,
+      width         : 450,
+      height        : 200,
+      resizable     : false,
+      draggable     : true,
+      closeOnEscape : false,
+      open          : function () {
+        $("#id-processing").dialog().parents(".ui-dialog").find(".ui-dialog-titlebar").remove();
+      }
+    });
+    $("#importProgressBar" ).progressbar({value:0});
+    $("#importProgressBar").show( );
+    pollLoop( );
```

Comment:
I think it would be better to do

    loop = window.setInterval(setIntermediate, 5)

here and delete the function `pollLoop`, and declare the `var loop` as the first line of this closure (line 29 just after the `CRM.$(function(){`) that way it is accessible from all functions and can be cleared by `setIntermediate` once progress is 100%.

Reply:
@colemanw I've pushed the changes as per the above suggestion. Thanks 👍